### ISSUE TITLE
Extremely slow mapping Record.into(ProxyableInterface.class)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#4772]
The following is currently rather slow in benchmarks:

```java
interface Proxyable {
}

for (int i = 0; i < 1000000; i++)
    record.into(Proxyable.class);
```

Benchmarks show extreme time wasting in jOOR's looking up of suitable methods:

![image](https://cloud.githubusercontent.com/assets/734593/11586115/18cdc4a8-9a71-11e5-9b54-a0fd68f5fbb6.png)

----
See also:
http://www.jooq.org/doc/latest/manual/sql-execution/performance-considerations/#comment-2390864218

**Comment by jlorenzen** [comment]
@lukaseder Thanks for looking into these so quickly

**Comment by lukaseder** [comment]
@jlorenzen No worries. They were pretty easy to fix and significant at the same time.

**Comment by jlorenzen** [comment]
@lukaseder has 3.8.0 been released yet or have these fixes been ported back to 3.5, 3.6, or 3.7?

**Comment by lukaseder** [comment]
3.8 is due for late Q1 2016. A 3.7 patch release should be delivered next week. 3.5 and 3.6 in early 2016, hopefully.

If this is urgent, I suggest applying the patch to your version:
https://github.com/jOOQ/jOOQ/commit/20d3e4114656cf9f298858ff029e8336954a9aaa

Hope this helps,
Lukas
